opencda/core/ml_libs/rl/envs/simple_carla_env.py

```diff
@@ -0,0 +1,695 @@
+        self._off_road_is_failure = self._cfg.off_road_is_failure                   # whether to consider running off road as failure
+        self._wrong_direction_is_failure = self._cfg.wrong_direction_is_failure     # whether to consider driving in wrong direction as failure
+        self._off_route_is_failure = self._cfg.off_route_is_failure                 # whether to consider driving off planned route as failure
+        self._off_route_distance = self._cfg.off_route_distance                     # distance to route threshold to mark current episode as failure
+        # reward config
+        self._reward_type = self._cfg.reward_type
+        assert set(self._reward_type).issubset(self._reward_type), \
+            set(self._reward_type)
+        self._success_distance = self._cfg.success_distance                         # distance to navigation target threshold to determine success
+        self._success_reward = self._cfg.success_reward                             # numerical reward for a successful navigation
+        self._max_speed = self._cfg.max_speed                                       # maximum allowed speed
+        self._collided = False                                                      # take penalty for collision
+        self._stuck = False                                                         # take penalty for stucking in traffic
+        self._ran_light = False                                                     # take penalty for running red light
+        self._off_road = False                                                      # take penalty for running off road
+        self._wrong_direction = False                                               # take penalty for driving in wrong direction
+        self._off_route = False                                                     # take penalty for driving off planned route
+        self._stuck_detector = StuckDetector(self._cfg.stuck_len)
+        # time configs
+        self._tick = 0
+        self._timeout = float('inf')
+        self._launched_simulator = False
+        # visualization configs
+        self._visualize_cfg = self._cfg.visualize
+        self._simulator_databuffer = dict()
+        self._visualizer = None
+        self._last_canvas = None
+
+        # import the action list
+        self._acc_list = self._cfg.ACC_LIST
+        self._steer_list = self._cfg.STEER_LIST
+
+        # configurate discrete or continuous env
+        self._env_is_discrete = False
+        self._env_is_continuous = False
+
+        # accumulated final reward
+        self._final_eval_reward = 0.0
+
+        # make sure the two settings are exclusive
+        if 'env_action_space' in self._cfg:
+            self._env_is_discrete = True if (
+                    self._cfg['env_action_space'] == 'discrete' and not self._env_is_continuous) else False
+            self._env_is_continuous = True if (
```

Comment:
does env_is_discrete has to be the opposite as continuous? if so, change the code here to env_is_continues = !_env_is_discrete